building.rb

```diff
@@ -1,2 +1,30 @@
 class Building
+  attr_accessor :address, :apartments
+  def initialize(args = {})
+    @address = args.fetch(:address, '')
+    @apartments = args.fetch(:apartments, [])
+  end 
+
+  def total_monthly_revenue
+    apartments.map! { |apartment| apartment.monthly_rent}
+    apartments.reduce(:+)
+  end
+
+  def total_room_count
+    apartments.map! { |apartment| apartment.room_count}
+    apartments.reduce(:+)
+  end
+
+  def apartments_by_rent
+    apartments.sort_by {|apartment| apartment.monthly_rent}.reverse
```

Comment:
Since monthly rent is a number, you can just make it negative to get your apartments sorted in the correct order without having to sort them and then call reverse on the sorted array.
```ruby
apartments.sort_by { |apartment| -apartment.monthly_rent }
```